File: spring-data-requery-java/src/test/java/com/coupang/springframework/data/requery/java/domain/blob/BinaryDataTest.java
package com.coupang.springframework.data.requery.java.domain.blob;

import com.coupang.springframework.data.requery.java.domain.AbstractDomainTest;
import org.junit.Test;

import static org.assertj.core.api.Assertions.assertThat;

/**
 * com.coupang.springframework.data.requery.java.domain.blob.BinaryDataTest
 *
 * @author debop
 * @since 18. 6. 4
 */
public class BinaryDataTest extends AbstractDomainTest {

    @Test
    public void saveBlobData() {
        byte[] bytes = new byte[8192];
        rnd.nextBytes(bytes);

        BinaryData binData = new BinaryData();
        binData.setName("binary data");
        binData.setPicture(bytes);

        requeryTemplate.insert(binData);
        assertThat(binData).isNotNull();

        BinaryData loaded = requeryTemplate.findById(BinaryData.class, binData.id);
        assertThat(loaded).isNotNull().isEqualTo(binData);
        assertThat(loaded.getPicture()).isNotNull().isEqualTo(binData.getPicture());

        rnd.nextBytes(bytes);
        loaded.setPicture(bytes);
        requeryTemplate.upsert(loaded);
    }
}

File: spring-data-requery-kotlin/src/test/kotlin/com/coupang/springframework/data/requery/domain/blob/BinaryDataTest.kt
package com.coupang.springframework.data.requery.domain.blob

import com.coupang.springframework.data.requery.domain.AbstractDomainTest
import org.assertj.core.api.Assertions.assertThat
import org.junit.Test

/**
 * com.coupang.springframework.data.requery.domain.blob.BinaryDataTest
 * @author debop
 * @since 18. 5. 31
 */
class BinaryDataTest: AbstractDomainTest() {


    @Test
    fun `deal blob property`() {
        val bytes = ByteArray(8192)
        rnd.nextBytes(bytes)

        val binData = BinaryData().apply {
            name = "binary data"
            picture = bytes
        }

        requeryTemplate.insert(binData)

        val loaded = requeryTemplate.findById(BinaryData::class.java, binData.id)!!
        assertThat(loaded.id).isEqualTo(binData.id)
        assertThat(loaded.picture).isNotNull().isEqualTo(bytes)
    }
}